### PR TITLE
[analysis_options] sync lint rules with flutter/flutter

### PR DESCRIPTION
**Pull request by goderbauer**
Syncs the lint configuration from the master file in the flutter/flutter repository over to this repository and fixes all resulting violations.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by goderbauer** [comment]
@stuartmorgan Is there anything special I need to do for packages that have a "NEXT" changelog entry? The check is not happy with what I did: https://cirrus-ci.com/task/5907580605169664

File: packages/standard_message_codec/example/lib/readme_excerpts.dart [hunk 1]
```diff
@@ -16,5 +16,5 @@ void main() {
     'foo': true,
     3: 'fizz',
   });
-  print('The encoded message is $data');
+  print('The encoded message is $data'); // ignore: avoid_print
```
Comment (stuartmorgan) [inline packages/standard_message_codec/example/lib/readme_excerpts.dart:19]: This should be at the file level so this doesn't show up in the excerpt (once the excerpt annotations are fixed).

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks, this has been on my TODO list for quite a while!

Just some small things here and there.

**Comment by stuartmorgan** [comment]
> @stuartmorgan Is there anything special I need to do for packages that have a "NEXT" changelog entry? The check is not happy with what I did: https://cirrus-ci.com/task/5907580605169664

You can't use `NEXT` if you are changing client-affecting code without overriding the check, because of [repo policy](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version). If there are cases you are absolutely sure shouldn't be versioned, we can override them, but at least some of them should be.

**Comment by goderbauer** [comment]
This is ready for another review.

For the packages `cross_file` and `standard_message_codec` I'd like to get an exception for versioning and changeloging as there are no meaningful changes to production code to these two packages.

File: packages/cross_file/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Fixed lint warnings in tests.
```
Comment (stuartmorgan) [inline packages/cross_file/CHANGELOG.md:3]: Nit: Fixes

Per style guide.

File: packages/standard_message_codec/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.0.1+2
+
+* Fixed lint warnings.
```
Comment (stuartmorgan) [inline packages/standard_message_codec/CHANGELOG.md:3]: Same.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nit. Thanks again!

**Pull request merged**